### DISCUSSION TITLE
Support for other devices

### DISCUSSION THREAD

**Discussion by Red4Hack** [kdrag0n/android-webinstall#9] [Ideas]
Would it be possible to support other devices?

I made an example in json:
```
{
    "devices": [
        {
            "manufacturerName": "",
            "url": "website to get unlock key",
            "steps": {
                "1": [
                    "command"
                ]

            }
        },
        {
            "name": "",
            "model": "",
            "steps": {
                "1": [
                    "command"
                ],
                "2": [
                    "command"
                ]
            }
        }
    ]
}
```

if the device is of the same brand it does those commands:
```
"manufacturerName": "Motorola Inc."
 "url": "https://motorola-global-portal.custhelp.com/app/standalone/bootloader/unlock-your-device-b",
            "steps": {
                "1": [
                    "fastboot oem get_unlock_data" 
                ],
                "2": [
                    "fastboot oem unlock $unlock_code"
               ] 
            }
```

the first step will return a key that will need to be redirected to this "url":
![image](https://user-images.githubusercontent.com/81523685/191892801-4fd95683-bddd-4ea3-bf33-e299ddd01c15.png)
![image](https://user-images.githubusercontent.com/81523685/191892983-0ec6672e-9432-4541-96c6-a633c37c3005.png)
after that you will receive an email with unlock code:
```
Unlock Code: insert code to unlock
```
and then continue the installation using android-webinstall
![image](https://user-images.githubusercontent.com/81523685/191893748-2d50852c-deb2-4da8-a01b-2d15ef821e72.png)

if you don't have a manufacturer, it unlocks by name or model

the installation part too, if possible using the same example done in json:
```
"manufacturerName": "Motorola Inc.",
"installsteps":{
  "1": [
      "fastboot flash recovery LineageOS_Recovery.img"
 ],
  "2":[
      "adb sideload LineageOS_18.1.zip"
  ]
}
```

**Comment by kdrag0n** [comment]
Structured JSON commands are relatively easy to implement, but if we're going in the direction of extensibility, I don't think a purely declarative config is enough to express the logic needed for some devices. Maybe a documented API for adding device support with JS would be better.

(This installer is due for a rewrite anyway, so until then I don't plan to work on this)
